guillotina_amqp/utils.py

```diff
@@ -115,12 +117,17 @@ async def add_task(func, *args, _request=None, _retries=3, _task_id=None, **kwar
                     "task_id": task_id,
                 }
             )
+
             # Publish task data on rabbitmq
+            dest_queue = (
+                app_settings["amqp"]["queue"] if dest_queue is None else dest_queue
+            )
+
             with watch_amqp("publish"):
                 await channel.publish(
                     data,
                     exchange_name=app_settings["amqp"]["exchange"],
-                    routing_key=app_settings["amqp"]["queue"],
+                    routing_key=dest_queue,
```

Comment:
If the queue doesn't exist, won't this message be discarded as soon as it's sent to the exchange?

Reply:
That is correct, but it would get created as soon as a worker spins up with the new queue in env.AMQP_QUEUE